src/MoBi.UI/Services/MoBiApplication.cs

```diff
@@ -56,5 +58,29 @@ protected override void OnRun()
             throw;
          }
       }
+
+      public bool PreFilterMessage(ref Message m)
+      {
+         if (!isWindowsKeyDownMessage(m))
+            return false;
+
+         switch (keyCodeFrom(m))
+         {
+            case (int)(Keys.Control | Keys.W):
+               _mainPresenter.ActivePresenter?.Close();
+               return true;
+         }
+         return false;
+      }
+
+      private static int keyCodeFrom(Message m)
+      {
+         return (int)m.WParam | (int)Control.ModifierKeys;
+      }
+
+      private static bool isWindowsKeyDownMessage(Message m)
+      {
+         return m.Msg == 256;
```

Comment:
damn!

Reply:
can we explain what the heck this means? 256?

Reply:
It means it's a key down message

Reply:
I don't think there's a built in enum or anything to translate windows messages in .net. this number 256 means what the method is named